apps/web/src/pages/Lesson.tsx

```diff
@@ -398,11 +398,12 @@ const Lesson: React.FC = () => {
             startAt: lesson.data.lesson.start,
           }}
           leave={() => {
-            session.leave();
-            sessionManager.leave();
+            session.leave(); // this shall close/stop related user media (video, share screen, etc)
+            sessionManager.leave(); // this shall close the connection (websocket), and invoke db mutations.
             if (lessonMembers.current.role === IUser.Role.Student) {
               const query = asRateLessonQuery({
                 lessonId: lessonMembers.current.lessonId,
+                lessonStart: lesson.data?.lesson.start || "",
```

Comment:
Let's add a safe-gaurd to return incase the lesson data is not yet available. 

```suggestion
                lessonStart: lesson.data.lesson.start,
```